README.md

```diff
@@ -114,6 +117,52 @@ module.exports = {
 }
 ```
 
+#### Hot Module Reloading (HMR)
+
+extract-mini-css-plugin supports hot reloading of actual css files in development. Some options are provided to enable HMR of both standard stylesheets and locally scoped css or CSS modules. Below is an example configuration of mini-css for HMR use with CSS modules
+
+**webpack.config.js**
+
+```js
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+module.exports = {
+  plugins: [
+    new MiniCssExtractPlugin({
+      // Options similar to the same options in webpackOptions.output
+      // both options are optional
+      filename: "[name].css",
+      chunkFilename: "[id].css"
+    })
+  ],
+  module: {
+    rules: [
+      {
+        test: /\.css$/,
+        use: [
+          {
+            loader: MiniCssExtractPlugin.loader,
+            options: {
+              // only enable hot in development
+              hot: process.env.NODE_ENV === 'development',
+              // modules enables hmr of css modules which
+              // need to be handled differently compared to typical css files
+              modules: true,
+              // in the event that HMR is not working correctly for your setup.
+              // a fallback option has been implemented to force css files to be reloaded
+              // reloadAll will force all stylesheets to be reloaded when receiving an HMR update
+              // this option is helpful when dealing with css-modules as they are harder to HMR
+              reloadAll: true,
```

Comment:
When it can be happens? I think we should add more documentation about this option. Maybe we can check this on client side and reload all if something broken/event not handled/or something other unexpected?

Reply:
Yeah this isnt an ideal one, but sometimes reloading **just wont work** this pretty much ensures reloading will work. 

I will remove this from my project as well... but it does offer an effective fallback

Reply:
@ScriptedAlchemy maybe you can find issues/examples where it really doesn't work? We can add this option after release and feedback from users, maybe we just have a bug and we can fix it without `reloadAll` :+1:

Reply:
The `reloadAll` option is for dealing with commons-chunk, eg [this example)(https://github.com/shepherdwind/css-hot-loader/blob/master/examples/commons-chunk/webpack.config.js#L24). Default the update logic is that

1. a.css required by a.js, and we have output a.js, then
2. when a.css update, a.js will receive update message, so the hmr module will reload a.js

But, a.css maybe extract to common.css, in this condition, we should reload all css file.